modules/network/gpu-rdma-vpc/variables.tf

```
@@ -0,0 +1,150 @@
+  type = object({
+    count       = number
+    name_prefix = string
+    ip_range    = string
+    region      = optional(string)
+  })
+  default = {
+    count       = 8
+    name_prefix = null
+    ip_range    = "192.168.0.0/16"
+    region      = null
+  }
+
+  validation {
+    condition     = var.subnetworks_template.count > 0
+    error_message = "Number of subnetworks must be greater than 0"
+  }
+
+  validation {
+    condition     = can(cidrhost(var.subnetworks_template.ip_range, 0))
+    error_message = "IP address range must be in CIDR format."
+  }
+}
+
+variable "network_routing_mode" {
+  type        = string
+  default     = "REGIONAL"
+  description = "The network routing mode (default \"REGIONAL\")"
+
+  validation {
+    condition     = contains(["GLOBAL", "REGIONAL"], var.network_routing_mode)
+    error_message = "The network routing mode must either be \"GLOBAL\" or \"REGIONAL\"."
+  }
+}
+
+variable "network_description" {
+  type        = string
+  description = "An optional description of this resource (changes will trigger resource destroy/create)"
+  default     = ""
+}
+
+variable "shared_vpc_host" {
+  type        = bool
+  description = "Makes this project a Shared VPC host if 'true' (default 'false')"
+  default     = false
+}
+
+variable "delete_default_internet_gateway_routes" {
+  type        = bool
+  description = "If set, ensure that all routes within the network specified whose names begin with 'default-route' and with a next hop of 'default-internet-gateway' are deleted"
+  default     = false
+}
+
+variable "enable_internal_traffic" {
+  type        = bool
+  description = "Enable a firewall rule to allow all internal TCP, UDP, and ICMP traffic within the network"
+  default     = true
+}
+
+variable "firewall_rules" {
+  type        = any
+  description = "List of firewall rules"
+  default     = []
+}
+
+variable "firewall_log_config" {
+  type        = string
+  description = "Firewall log configuration for Toolkit firewall rules (var.enable_iap_ssh_ingress and others)"
+  default     = "DISABLE_LOGGING"
+  nullable    = false
+
+  validation {
+    condition = contains([
+      "INCLUDE_ALL_METADATA",
+      "EXCLUDE_ALL_METADATA",
+      "DISABLE_LOGGING",
+    ], var.firewall_log_config)
+    error_message = "var.firewall_log_config must be set to \"DISABLE_LOGGING\", or enable logging with \"INCLUDE_ALL_METADATA\" or \"EXCLUDE_ALL_METADATA\""
+  }
+}
+
+variable "network_profile" {
+  description = <<-EOT
+  A full or partial URL of the network profile to apply to this network.
+  This field can be set only at resource creation time. For example, the
+  following are valid URLs:
+  - https://www.googleapis.com/compute/beta/projects/{projectId}/global/networkProfiles/{network_profile_name}
+  - projects/{projectId}/global/networkProfiles/{network_profile_name}}
+  EOT
+  type        = string
+  nullable    = false
+
+  validation {
+    condition     = var.network_profile != ""
```

This is a little safer version in case `nullable = false` is ever removed.

```suggestion
    condition     = can(coalesce(var.network_profile))
```